rust/private/rust.bzl

```diff
@@ -219,6 +219,100 @@ def _rust_binary_impl(ctx):
         ),
     )
 
+def _create_test_launcher(ctx, toolchain, output, providers):
```

Comment:
Do we need to do all of this? What I had in mind was to add this into rust_test rule declaration:

```rust
"_process_wrapper": attr.label(
        default = Label("//util/test_runner"),
        executable = True,
        allow_single_file = True,
        cfg = "exec",
    ),
```

And then:
0) add a rust_binary at //util/test_runner:test_runner
1) in rust_test access the runner using `ctx.executable._test_runner`
2) put that in the DefaultInfo that rust_test provides
3) register action that writes list of env variables into a file
4) put that file into DefaultInfo runfiles
5) register action that symlinks the actual test binary into some path
6) put both the test and the symlink into runfiles
7) modify test_runner to look into runfiles for env-vars file and for the symlink

Does it make sense?

Reply:
This is different from the process wrapper because the process wrapper is used as an executable for an action. However, the launcher/test_runner needs to be the executable returned by the rule so it gets executed when you `bazel test //:my_target`. Bazel will otherwise throw an error if you return an executable that is not created by an action of that rule. Is there another way around this? Aside from having to create the binary in an action in the rule, I what you've written is more or less what I'm doing, right?